Mathlib/Analysis/Meromorphic/Divisor/Basic.lean

```diff
@@ -282,31 +208,33 @@ instance : Min (DivisorOn U) where
     supportDiscreteWithinDomain' := by
       filter_upwards [D₁.supportDiscreteWithinDomain, D₂.supportDiscreteWithinDomain]
       intro _ h₁ h₂
-      simp [h₁, h₂]
-  }
+      simp [h₁, h₂] }
 
-/-- Helper lemma for the `simp` tactic: the function associated with the max of
-two divisors is the pointwise max of the associated functions. -/
 @[simp]
-lemma min_fun {D₁ D₂ : DivisorOn U} {x : 𝕜} : min D₁ D₂ x = min (D₁ x) (D₂ x) := rfl
+lemma min_def {D₁ D₂ : DivisorOn U} {x : 𝕜} : min D₁ D₂ x = min (D₁ x) (D₂ x) := rfl
 
-/-- Divisors form a lattice. -/
 instance : Lattice (DivisorOn U) where
   le := (· ≤ ·)
-  le_refl := by simp
-  le_trans := by exact fun D₁ D₂ D₃ h₁₂ h₂₃ x ↦ (h₁₂ x).trans (h₂₃ x)
-  le_antisymm := by
-    intro D₁ D₂ h₁₂ h₂₁
+  lt := (· < ·)
+  le_refl := by simp [le_def]
+  le_trans D₁ D₂ D₃ h₁₂ h₂₃ := fun x ↦ (h₁₂ x).trans (h₂₃ x)
+  le_antisymm D₁ D₂ h₁₂ h₂₁ := by
     ext x
     exact Int.le_antisymm (h₁₂ x) (h₂₁ x)
-  sup := (max · ·)
-  le_sup_left := fun D₁ D₂ x ↦ by simp
-  le_sup_right := fun D₁ D₂ x ↦ by simp
-  sup_le := fun D₁ D₂ D₃ h₁₃ h₂₃ x ↦ by simp [h₁₃ x, h₂₃ x]
-  inf := (min · ·)
-  inf_le_left := fun D₁ D₂ x ↦ by simp
-  inf_le_right := fun D₁ D₂ x ↦ by simp
-  le_inf := fun D₁ D₂ D₃ h₁₃ h₂₃ x ↦ by simp [h₁₃ x, h₂₃ x]
+  sup := max
+  le_sup_left D₁ D₂ := fun x ↦ by simp
+  le_sup_right D₁ D₂ := fun x ↦ by simp
+  sup_le D₁ D₂ D₃ h₁₃ h₂₃ := fun x ↦ by simp [h₁₃ x, h₂₃ x]
+  inf := min
+  inf_le_left D₁ D₂ := fun x ↦ by simp
+  inf_le_right D₁ D₂ := fun x ↦ by simp
+  le_inf D₁ D₂ D₃ h₁₃ h₂₃ := fun x ↦ by simp [h₁₃ x, h₂₃ x]
+
+/-- Divisors form an ordered commutative group -/
+instance : OrderedAddCommGroup (DivisorOn U) where
```

Comment:
By moving this to the end, we can reuse the work on `AddCommGroup` and `Lattice` above.

Reply:
Roger.